### PR TITLE
`Development`: Improve detail overview list client tests

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
There was an issue in programming exercise detail pages, because submission-policy.service inserted null values, which have not been typed. The typing has been adjusted #8028, this PR adds some tests, which should have recognized this issue

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Programming Exercise Detail
3. Verify that the page behaves normal (consistency check works, repo clone works, git diff works)
4. Verify that also the last Detail (Build log statistics is displayed)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **Refactor**
	- Enhanced type safety in the detail overview list component by refining variable declarations and removing unnecessary type assertions.

### CODE REVIEW

File: src/test/javascript/spec/component/detail-overview-list.component.spec.ts [hunk 1]
```diff
@@ -62,6 +62,21 @@
...
+        fixture.detectChanges();
+        const detailListTitleDOMElements = fixture.nativeElement.querySelectorAll('dt[id^=detail-title]');
+        expect(detailListTitleDOMElements).toHaveLength(1);
+        const titleDetailTitle = fixture.nativeElement.querySelector('dt[id=detail-title-title]');
+        const titleDetailValue = fixture.nativeElement.querySelector('dd[id=detail-value-title]');
+        expect(titleDetailTitle).toBeDefined();
+        expect(titleDetailValue).toBeDefined();
+        expect(titleDetailTitle.textContent).toContain('title');
+        expect(titleDetailValue.textContent).toContain('A Title');
+    });
```
[the remark this answers is not in the document]
Comment (milljoniaer) [reply src/test/javascript/spec/component/detail-overview-list.component.spec.ts:78]: yes that is ensured by checking that the last detail and only the last detail (as the other ones are falsy) is still rendered

**Review by Gusti2010 — APPROVED** [review]
Manually tested on ts1. Everything works as expected!

**Review by RY997 — APPROVED** [review]
Tested on ts9, worked as expected. And code also looks good to me.

**Review by Gusti2010 — APPROVED** [review]
reapprove

**Review by Predixx — APPROVED** [review]
Tested in ts2. Looks good to me

**Review by RY997 — APPROVED** [review]
Reapprove

**Review by nata-81 — APPROVED** [review]
Tested on ts1 works as expected.

**Pull request merged**